### PR TITLE
Use sysUpTime to calculate and log interface state change times

### LINKED ISSUE #77: Use `ifLastChange` to calculate link state change times

### PR DESCRIPTION
**Pull request by lunkwill42**
Fixes #77

### CODE REVIEW

File: src/zino/tasks/linkstatetask.py [hunk 1]
```diff
@@ -39,10 +40,13 @@ class LinkStateTask(Task):
     case: Descriptions are mandated for both physical ports and their sub-units.
     """
 
+    sysuptime: Optional[int] = 0
```
Comment (hmpf) [inline src/zino/tasks/linkstatetask.py:43]: What's the difference between sysuptime set to 0 and None?

**Review by hmpf — COMMENTED** [review]
See inline question.

File: src/zino/tasks/linkstatetask.py (hunk 1 above)
Comment (lunkwill42) [reply src/zino/tasks/linkstatetask.py:43]: `None` would usually mean that we don't know the `sysUpTime` value, while `0` would mean that the device booted *just now*.

However, the `Optional` here seems to be redundant. Within this plugin, the value isn't optional.  The original link state code basically defaults to using 0 as the base for calculating timestamps if the sysuptime value couldn't be fetched.

File: src/zino/statemodels.py [hunk 2]
```diff
@@ -80,6 +80,13 @@ def is_cisco(self):
     def is_juniper(self):
         return self.enterprise_id == 2636
 
+    def set_boot_time_from_uptime(self, uptime: int):
+        """Calculates and sets the device boot time from a current uptime value.
+
+        :param uptime: An uptime value in 100ths of a second
+        """
+        self.boot_time = datetime.datetime.now() - datetime.timedelta(seconds=uptime / 100)
```
Comment (lunkwill42) [inline src/zino/statemodels.py:88]: Actually, I would change this slightly, now that I see it again in hindsight.

We already have a utility function to ensure we generate timestamps with timezone info: `zino.time.now()`

**Review by hmpf — APPROVED** [review]

**Pull request merged**